nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -562,11 +562,9 @@ sub addProfile {
             next;
         }
         my $date = strftime("%F", localtime(lstat($link)->mtime));
-        my $version =
-            -e "$link/nixos-version"
-            ? readFile("$link/nixos-version")
-            : basename((glob(dirname(Cwd::abs_path("$link/kernel")) . "/lib/modules/*"))[0]);
-        addEntry("NixOS - Configuration " . nrFromGen($link) . " ($date - $version)", $link);
+        my $nixos_version = readFile("$link/nixos-version");
+        my $kernel_version = basename((glob(dirname(Cwd::abs_path("$link/kernel")) . "/lib/modules/*"))[0]);
```

Comment:
Can we do something like @lheckemann suggested here https://github.com/NixOS/nixpkgs/pull/114351#issuecomment-1022056117?

Reply:
That seems more fragile but if @lheckemann thinks it's the better idea (just give this comment a thumbs up), I'll change and retest.
The reason I did it this way is that it was already implemented like this and systemd-boot does pretty much the same.

Reply:
Well it does feel a little bit more fragile, in that it restricts the kernel naming convention.
On the other hand using `lib/modules` currently prevents people that compile their kernel without modules from using the nixos boot loader machinery (because that directory will then not exist).

Reply:
Perhaps the approach from #114351, with a fallback to the store path, would be best? I'm quite undecided.

Reply:
I like that fallback idea. It gives users like me who set their `LOCALVERSION` that string in the grub menu and also gives people who build without module a way to boot